sdcm/sla/sla_tests.py

```diff
@@ -0,0 +1,152 @@
+                                                                                   db_cluster=tester.db_cluster)
+                # TODO: add validation
+                LOGGER.debug('RUN TIME PER ROLE BEFORE INCREASE: {}'.format(runtime_per_role_before_increase))
+            except Exception as details:  # pylint: disable=broad-except
+                wp_event.add_error([str(details)])
+                wp_event.full_traceback = traceback.format_exc()
+                wp_event.severity = Severity.ERROR
+
+    # pylint: disable=too-many-arguments
+    def alter_sl_and_validate_scheduler_runtime(self, tester, service_level, new_shares, read_users, prometheus_stats,
+                                                sleep=600):
+        # pylint: disable=not-context-manager
+        with WorkloadPrioritizationTestEvent(step=f"Alter shares from {service_level.shares} to {new_shares} Service "
+                                                  f"Level {service_level.name} and validate scheduler runtime "
+                                                  f"during load") as wp_event:
+            try:
+                service_level.alter(new_shares=new_shares)
+                start_time = time.time() + 60
+                time.sleep(sleep)
+                end_time = time.time()
+                runtime_per_role_before_increase = self.validate_scheduler_runtime(start_time=start_time,
+                                                                                   end_time=end_time,
+                                                                                   read_users=read_users,
+                                                                                   prometheus_stats=prometheus_stats,
+                                                                                   db_cluster=tester.db_cluster)
+                # TODO: add validation
+                LOGGER.debug('RUN TIME PER ROLE AFTER INCREASE: {}'.format(runtime_per_role_before_increase))
+            except Exception as details:  # pylint: disable=broad-except
+                wp_event.add_error([str(details)])
+                wp_event.full_traceback = traceback.format_exc()
+                wp_event.severity = Severity.ERROR
+
+
+class SlaTests(Steps):
+    STRESS_READ_CMD = 'cassandra-stress read cl=ALL duration={duration} -mode cql3 native user={user} ' \
+                      'password={password} -rate threads={threads} -pop {pop}'
+
+    def test_increase_shares_during_load(self, tester, prometheus_stats):
+        low_share = 20
+        high_share = 500
+
+        with tester.db_cluster.cql_connection_patient(node=tester.db_cluster.nodes[0],
+                                                      user=DEFAULT_USER,
+                                                      password=DEFAULT_USER_PASSWORD) as session:
+            role_low = create_sla_auth(session=session, shares=low_share, index=0)
```

Comment:
we need the session for those two command, why we keep it open for the rest of the test ?

Reply:
oh my, the roles object holds a reference to session ? I think this hidden detail gonna bite us